control/nlsys.py

```diff
@@ -2499,17 +2503,15 @@ def interconnect(
             elif not found_system:
                 raise ValueError("could not find signal %s" % sname)
         else:
-            # Regular signal specification
-            if not isinstance(connection, list):
-                dprint(f"  converting item to list")
-                connection = [connection]
-            for spec in connection:
+            # Utility function to find named output or input signal
+            def _find_output_or_input_signal(spec):
+                signal_list = []
                 try:
                     # First trying looking in the output signals
                     osys, indices, gain = _parse_spec(syslist, spec, 'output')
                     for osig in indices:
-                        dprint(f"  adding output {(osys, osig, gain)}")
-                        new_outlist.append((osys, osig, gain))
+                            dprint(f"  adding output {(osys, osig, gain)}")
+                            signal_list.append((osys, osig, gain))
```

Comment:
```suggestion
                        dprint(f"  adding output {(osys, osig, gain)}")
                        signal_list.append((osys, osig, gain))
```
There is an extra indentation here. It does not affect correctness, but I guess it was accidental.